psi4/driver/driver_cbs.py

```diff
@@ -701,13 +936,170 @@ def return_energy_components():
+            scf["options"] = False
+            metadata.append(scf)
+    # 2c) keep processing current stage
+        stage["treatment"] = item.get("treatment", "scf" if len(metadata) == 0 else "corl")
+        stage["stage"] = item.get("stage", False)
+        if not stage["stage"]:
+            if len(metadata) == 0:
+                stage["stage"] = "scf"
+            elif len(metadata) == 1:
+                stage["stage"] = "corl"
+            else:
+                stage["stage"] = "delta{0:d}".format(len(metadata) - 1)
+        stage["scheme"] = item.get("scheme", _get_default_xtpl(len(stage["basis"][1]), stage["treatment"]))
+        if len(metadata) > 0:
+            stage["wfn_lo"] = item.get("wfn_lo", metadata[-1].get("wfn")).lower()
+            stage["basis_lo"] = _expand_bracketed_basis(item.get("basis_lo", item["basis"]).lower())
+            if len(stage["basis"][0]) != len(stage["basis_lo"][0]):
+                raise ValidationError("""Number of basis sets inconsistent
+                                            between high ({0:d}) and low ({1:d}) levels.""".format(
+                    len(stage["basis"][0]), len(stage["basis_lo"][0])))
+        stage["alpha"] = item.get("alpha", None)
+        stage["options"] = item.get("options", False)
+        metadata.append(stage)
+    return (metadata)
+
+
+def _process_cbs_kwargs(kwargs):
+    """ A helper function which translates supplied kwargs into the
+    ``cbs_metadata`` format and passes it for validation.
+
+    Parameters
+    ----------
+    kwargs : dict
+        kwargs containing the CBS function specification.
+
+    Returns
+    -------
+    array
+        Array of dictionaries, with each item consisting of an extrapolation
+        stage. All validation takes place here.
+    """
+
+    if "cbs_metadata" in kwargs:
+        # if we passed in a dict, validate it right away
+        cbs_metadata = kwargs.get("cbs_metadata")
+    else:
+        # if we passed in options, check for consecutive correlations first
+        if "delta_wfn" in kwargs and "corl_wfn" not in kwargs:
+            raise ValidationError("Delta function supplied without corl_wfn defined.")
+        if "delta2_wfn" in kwargs and "delta_wfn" not in kwargs:
+            raise ValidationError("Second delta function supplied without delta_wfn defined.")
+        cbs_metadata = []
+        possible_stages = ["scf", "corl"]
+        while len(possible_stages) > 0:
+            sn = possible_stages.pop(0)
+            # either both *_wfn and *_basis have to be specified
+            if "{:s}_wfn".format(sn) in kwargs and "{:s}_basis".format(sn) in kwargs:
+                stage = {"wfn": kwargs["{:s}_wfn".format(sn)], "basis": kwargs["{:s}_basis".format(sn)]}
+            # or we're at a scf stage which can be implied with a provided scf_basis
+            elif sn == "scf" and "{:s}_basis".format(sn) in kwargs:
+                stage = {"wfn": "hf", "basis": kwargs["{:s}_basis".format(sn)]}
```

Comment:
Can this handle the internal setup for SCF stage right `energy(cbs, corl_wfn='mp2', corl_basis='cc-pv[dt]z')`? I'm not seeing it, but maybe I'm wrong. You've got the running code handy.

Reply:
Yup, exactly the same behaviour as in 1.3a1:
[cbs_dict](https://github.com/psi4/psi4/files/2519138/post.txt), [1.3a1](https://github.com/psi4/psi4/files/2519139/pre.txt)